### PR TITLE
[W4.4b][F09-A1] Nicholas Ang

### PR DESCRIPTION
**Pull request by nicholasangcx**
Split address into different classes

### CODE REVIEW

**Comment by ng95junwei** [comment]
Hi Nicholas, 

it looks like your PR has failed the CI check because your tests are not passing. Perhaps it would be a good idea to look through and check to make sure all your unit tests pass.

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -8,10 +8,15 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, some street, some unit, some postal code";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in (block, street, unit, postal code) format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
+    Block block;
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/data/person/Address.java:15]: Assign ;private' accessibility to the variables.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -22,13 +27,23 @@
...
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
     }
 
+    private void splitAddress(String trimmedAddress) {
+        String delimiter = ",";
+        String[] tokens = trimmedAddress.split(delimiter);
+        this.block = new Block(tokens[0]);
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/data/person/Address.java:41]: Use meaningful variable names other than tokens[0] etc

File: src/seedu/addressbook/data/person/Unit.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package seedu.addressbook.data.person;
```
Comment (RonakLakhotia) [inline src/seedu/addressbook/data/person/Unit.java:1]: Include header comments at the start of each class.

**Review by RonakLakhotia — COMMENTED** [review]
Close the PR after reading the comments. Keep in mind to update the documentation after each change. Good work

**Pull request closed without merging**